src/providers/data_provider/dp_target_id.c

```diff
@@ -185,6 +212,176 @@ static void dp_req_initgr_pp(const char *req_name,
+              SYSDB_SESSION_RECORDING, sss_strerror(ret));
+        goto done;
+    }
+
+    /* Format output username */
+    name = sss_nss_get_name_from_msg(ctx->domain_info, res->msgs[0]);
+    ret = sss_nss_output_name(tmp_ctx, ctx->domain_info, name,
+                              be->override_space, &output_name);
+    if (ret != EOK) {
+        DEBUG(SSSDBG_CRIT_FAILURE,
+              "Failed formatting output username from \"%s\": %s\n",
+              name, sss_strerror(ret));
+        goto done;
+    }
+
+    /* For each user name in session recording config */
+    conf_user = be->sr_conf.users;
+    if (conf_user != NULL) {
+        for (; *conf_user != NULL && !enabled; conf_user++) {
+            /* If it matches the requested user name */
+            if (strcmp(*conf_user, output_name) == 0) {
+                enabled = true;
+            }
+        }
+    }
+
+    /* If we have groups in config and are not yet enabled */
+    if (be->sr_conf.groups != NULL &&
+        be->sr_conf.groups[0] != NULL &&
+        !enabled) {
+        /* For each group in response */
+        for (i = 0; i < res->count && !enabled; i++) {
+            /* Get the group msg */
+            if (i == 0) {
+                gid_t gid;
+                const char *attrs[] = { SYSDB_NAME, NULL };
+                /* Get the primary group */
+                gid = ldb_msg_find_attr_as_uint64(res->msgs[i], SYSDB_GIDNUM, 0);
+                if (gid == 0) {
+                    continue;
+                }
+                ret = sysdb_search_group_by_gid(tmp_ctx, ctx->domain_info,
+                                                gid, attrs, &msg);
+                if (ret == ENOENT) {
+                    continue;
+                } else if (ret != EOK) {
+                    DEBUG(SSSDBG_CRIT_FAILURE,
+                          "Failed retrieving group #%llu: %s\n",
+                          (unsigned long long)gid, sss_strerror(ret));
+                    goto done;
+                }
+            } else {
+                msg = res->msgs[i];
+            }
+            /* Get the group's output name */
+            name = sss_nss_get_name_from_msg(ctx->domain_info, msg);
+            if (name == NULL) {
+                continue;
+            }
+            ret = sss_nss_output_name(tmp_ctx, ctx->domain_info,
+                                      name, be->override_space,
+                                      &output_name);
+            if (ret != EOK) {
+                DEBUG(SSSDBG_CRIT_FAILURE,
+                      "Failed formatting output group name from \"%s\": %s\n",
+                      name, sss_strerror(ret));
+                goto done;
+            }
+            /* For each group in configuration */
+            for (conf_group = be->sr_conf.groups;
+                 *conf_group != NULL && !enabled;
+                 conf_group++) {
+                if (strcmp(*conf_group, output_name) == 0) {
```

Comment:
The "output_name" here doesn't reflect local overrides and shows the original group names.